# Elementos de Máquinas 2 – Projeto de mola

Criado por: Luan Henrique de Oliveira e Paola Fernanda Monteiro

• O mesmo deve ser entregue com todo o memorial de cálculos e justificativas de dados adotados durante a execução do mesmo.                    
• Os dados tabelados devem ter indicação de qual tabela foram retirados.                                                                          
• Os ganchos da mola também devem constar no projeto. 

Projete uma mola de tração helicoidal de vida infinita com extremidades de laço completo 
para uma carga mínima de 700 N e máxima de 1000 N, com um estiramento de 25 mm. A 
mola é para um equipamento de serviço de alimento e, portanto, deve ser feita de aço 
inoxidável. A frequência forçante é de 500 rpm. Considere um fator de projeto para fadiga 
de nf = 2

In [4]:
fmin = 700  #N
fmax = 1000 #N
y = 25      #mm 
n = 500     #rpm
nf = 2

C: Índice da mola                                                  
D: Diâmetro maior                                                  
d: Diâmetro menor                                                  

$ 4 \leq C \leq 12 $

$$ C = \frac{D}{d} $$

Supondo os valores iniciais de $C=9$ e $d=9$

In [5]:
C, d = 9, 9

In [6]:
def D(C, d):
    return C*d

D(C, d)

81

$$ D = Cd = 81 $$

Valor apropriado da tensão inicial de espira $ \tau_{i} $
$$ \tau_{i,1} = -4,231C^3 + 181,5C^2 - 3387C + 28640$$

In [7]:
def ti1(C):
    return round(-4.231*(C**3) + 181.5*(C**2) - 3387*C + 28640, 3)

ti1(C), round(ti1(C)/145, 3) 

(9774.101, 67.408)

$$ \tau_{i,1} = 9774,101 psi \approx 67,407 MPa $$

$$ \tau_{i,2} = -2,987C^3 + 139,7C^2 - 3427C + 38404 $$

In [8]:
def ti2(C):
    return round(-2.987*(C**3) + 139.7*(C**2) - 3427*C + 38404, 3)

ti2(C), round(ti2(C)/145, 3)

(16699.177, 115.167)

$$ \tau_{i, 2} = 16699,177 psi \approx 115,167 MPa $$

$$ \tau_{i} = \frac{\tau_{i, 1} + \tau_{i, 1}}{2} $$

In [9]:
def ti(C):
    return ti1(C) + ti2(C)

ti_N = round(ti(C)/145, 3)   #Tensão em Newton

ti(C), ti_N

(26473.278, 182.574)

$$ \tau_{i} = 26473,278 psi \approx 182,574 MPa $$

$ K_{s} $: Fator de cisalhamento direto
$$ K_{s} = 1 + \frac{0,5}{C} $$

In [10]:
def ks(C):
    return round(1 + 0.5/C, 3)

ks(C)

1.056

$$ K_{s} = 1,056 $$

$ F_{i} $: Força de tração inicial
$$ F_{i} = \frac{\pi d^3 \tau_{i}}{8K_{s}D} $$

In [11]:
from numpy import pi

In [12]:
def fi(C, d):
    return round(pi*ti_N*(d**3)/(8*ks(C)*D(C, d)), 3)

fi(C, d)

611.051

$ F_{i} = 611,051 N < F_{min}$. Não defletirá a mola.

$ F_{a} $: Força alternada
$$ F_{a} = \frac{F_{max} - F_{min}}{2} $$

In [13]:
def fa():
    return (fmax - fmin)/2

fa()

150.0

$$ F_{a} = 150 N $$

$ F_{m} $: Força média
$$ F_{m} = \frac{F_{max} + F_{min}}{2} $$

In [14]:
def fm():
    return (fmax + fmin)/2

fm()

850.0

$$ F_{m} = 850 N $$

$ \tau_{m} $: Tensão média
$$ \tau_{m} = K_{s} \frac{8F_{m}D}{\pi d^3} $$

In [15]:
def tm(C, d):
    return round(ks(C)*8*fm()*D(C, d)/(pi*d**3), 3)

tm(C, d)

253.969

$$ \tau_{m} = 253,969 MPa $$


$ K_{w} $: Fator de Wahl
$$ K_{w} = \frac{4C - 1}{4C - 4} + \frac{0.615}{C} $$

$ \tau_{a} $: Tensão de cisalhamento alternada
$$ \tau_{a} = K_{w} \frac{8F_{a}D}{\pi d^3} $$

In [16]:
def kw(C):
    return ( (4*C - 1)/(4*C - 4) ) + (0.615)/C

def ta(C):
    return round(kw(C)*8*fa()*D(C, d)/(pi*d**3), 3)

ta(C)

49.32

$$ \tau_{a} = 49,32 MPa $$

$ S_{ut} $: Limite de resistência a tração
$$ S_{ut} = A/d^m $$
$ S_{us} $: Limite de resistência a cisalhamento
$$ S_{us} = 0,667S_{ut} $$
$ S_{ys} $: Limite de resistência a torção
$$ S_{ys} = 0,45S_{ut} $$

![Constantes](ctes_material.png "Constantes")                                                                                       
Fonte: Design Handbook, 1987, p.19.

In [17]:
def sut(A, m):
    return round(A/(d**m), 3)

A = 1867 #MPa*mm^m
m = 0.146

sut(A, m)

1354.643

$$ S_{ut} = 1354,643 MPa $$

In [18]:
def sus(A, m):
    return round(0.667*sut(A, m), 3)

sus(A, m)

903.547

$$ S_{us} = 903,547 MPa $$

In [19]:
def sys(A, m):
    return round(0.45*sut(A, m), 3)

sys(A, m)

609.589

$$ S_{ys} = 609,589 MPa$$

$ S_{es} $: limite de resistência à fadiga em condições alternadas
$$ S_{es} = 0,5 \frac{S_{ew}S_{us}}{S_{us} - 0,5S_{ew}} $$

In [20]:
sew = 310.26 #MPa

def ses(A, m):
    return round(0.5*(sew*sus(A, m))/(sus(A, m) - 0.5*sew), 3)

ses(A, m)

187.285

$$ S_{es} = 187,285 MPa$$

$N_{fs}$: Coeficiente de segurança para espiras em torção
$$ N_{fs} = \frac{S_{es}(S_{us} - \tau_{min})}{S_{es}(\tau_{m} - \tau_{min}) + S_{us}\tau_{a}} $$

$ \tau_{min} $: Tensão de cisalhamento mínima 
$$ \tau_{min} = K_{B} \frac{8F_{min}D}{\pi d^3} $$

In [21]:
def tmin(fmin, C):
    kb = (4*C + 2)/(4*C - 3)
    return kb*8*fmin*D(C, d)/(pi*d**3)

In [22]:
def nfs(A, m, C, d, fmin):
    return round(ses(A, m)*(sus(A, m) - tmin(fmin, C))/(ses(A, m)*(tm(C, d) - tmin(fmin, C)) + sus(A, m)*ta(C)), 3)

nfs(A, m, C, d, fmin)

2.56

$$ N_{fs} = 2,56 $$

$$ C_{1} = \frac{2R_{1}}{d} = \frac{2D}{2d} = C $$

In [23]:
def C1():
    return C

$$ K_{b} = \frac{4C_1^2 - C_1 - 1}{4C_1(C_1 - 1)} $$ 

In [24]:
def kb():
    return round(4*((C1()**2) - C1() - 1)/(4*C1()*(C1() - 1)), 3)

kb()

0.986

$$ K_{b} = 0,986 $$

$$ \sigma_{a} = k_{b} \frac{16D F_a}{\pi d^3} + \frac{4F_{a}}{\pi d^2} $$

In [25]:
def sa():
    return round(kb()*( (16*D(C, d)*fa())/(pi*d**3) ) + 4*fa()/(pi*d**2), 3)

sa()

86.052

$$ \sigma_{a} = 86,052 MPa $$

$$ \sigma_{m} = k_{b} \frac{16D F_{m}}{\pi d^3} + \frac{4F_{m}}{\pi d^2} $$

In [26]:
def sm():
    return round(kb()*( (16*D(C, d)*fm())/(pi*d**3) ) + 4*fm()/(pi*d**2), 3)

sm()

487.629

$$ \sigma_{m} = 487,629 MPa $$

$$ \sigma_{min} = k_{b} \frac{16D F_{min}}{\pi d^3} + \frac{4F_{min}}{\pi d^2} $$

In [27]:
def smin():
    return round(kb()*( (16*D(C, d)*fmin)/(pi*d**3) ) + 4*fmin/(pi*d**2), 3)

smin()

401.577

$$ \sigma_{min} = 401,577 MPa $$

$S_{e}$: Resistência a endurança total
$$ S_{e} = \frac{S_{es}}{0,67} $$

In [28]:
def se(A, m):
    return round(ses(A, m)/0.67, 3)

se(A, m)

279.53

$$ S_{e} = 279,51 MPa $$

$$ N_{fb} = \frac{S_{e}(S_{ut} - \sigma_{min})}{S_{e}(\sigma_{m} - \sigma_{min}) + S_{ut}\sigma_{a}} $$

In [29]:
def nfb():
    return round(se(A, m)*(sut(A, m) - smin())/(se(A, m)*(sm() - smin()) + sut(A, m)*sa()), 3)

nfb()

1.894

$$ N_{fb} = 1,894 $$

$$ R_{2} = \frac{C_{2}d}{2} $$
Supondo inicialmente um valor de $C_{2} = 2$, temos:

In [30]:
C2 = 5

In [31]:
def r2():
    return round(C2*d/2, 3)

r2()

22.5

$$ R_{2} = 22.5 $$

$$ K_{w,2} = \frac{4C_{2} - 1}{4C_{2} - 4} $$

In [32]:
def kw2():
    return round((4*C2 - 1)/(4*C2 - 4), 3)

kw2()

1.188

$$ K_{w,2} = 1,188 $$

$\tau_{B}$: Tensão de torção no gancho

$$ \tau_{B_a} = K_{w,2} \frac{8D F_{a}}{\pi d^3} $$

In [33]:
def tba():
    return round(kw2()*8*D(C, d)*fa()/(pi*(d**3)), 3)

tba()

50.42

$$ \tau_{B_a} = 50,42MPa $$

$$ \tau_{B_m} = K_{w,2} \frac{8D F_{m}}{\pi d^3} $$

In [34]:
def tbm():
    return round(kw2()*8*D(C, d)*fm()/(pi*(d**3)), 3)

tbm()

285.715

$$ \tau_{B_m} = 285,715 MPa$$

$$ \tau_{B_{min}} = K_{w,2} \frac{8D F_{min}}{\pi d^3} $$

In [35]:
def tbmin():
    return round(kw2()*8*D(C, d)*fmin/(pi*(d**3)), 3)

tbmin()

235.295

$$ \tau_{B_{min}} = 235,295 MPa$$

$N_{fs}$: coeficiente de segurança à fadiga
$$ N_{fs} = \frac{S_{es}(S_{us} - \tau_{min})}{S_{es}(\tau_{m} - \tau_{min}) + S_{us} \tau_{a}} $$

In [36]:
ta(C)

49.32

In [37]:
def nfsb():
    return round(ses(A, m)*(sus(A, m) - tmin(fmin, C))/( ses(A, m)*(tm(C, d) - tmin(fmin, C)) + sus(A, m)*ta(C) ), 3)


nfsb()

2.56

$$ N_{fs_{B}} = 2,56 $$

$k$: constante da mola
$$ k = \frac{\Delta F}{y} $$

In [38]:
def k():
    return (fmax - fmin)/y

k()

12.0

$$ k = 12 N/mm$$

$N_{a}$: número de espiras ativas                                                                                                                      
$G$: Módulo de elasticidade transversal
$$ N_{a} = \frac{d^4 G}{8 D^3 k} $$

In [39]:
E = 193*10**3   #MPa
v = 0.265       #coef de poisson
G = E/(2*(1+v))

def na():
    return round((d**4)*G/(8*k()*(D(C, d)**3)), 0)

na()

10.0

$$ N_{a} \approx 10 espiras $$

$N_{t}$: Número total de espiras no corpo
$$ N_{t} = N_{a} + 1 $$

In [40]:
def nt():
    return na() + 1

nt()

11.0

$$ N_{t} = 11 espiras$$

$L_{b}$: Comprimento do corpo
$$ L_{B} = N_{t}d $$

In [41]:
def lb():
    return nt()*d

lb()

99.0

$$ L_{b} = 99mm $$

$L_{f}$: Comprimento livre
$$ L_{f} = L_{b} + 2L_{gancho} $$

In [42]:
def lf():
    di = D(C, d) - d
    return lb() + 2*di

lf()

243.0

$$ L_{f} = 243 mm $$

$y_{max}$: Deflexão máxima
$$ y_{max} = \frac{F_{max} - F_{min}}{k} $$

In [43]:
def ymax():
    return (fmax - fmin)/k()

ymax()

25.0

$$ y_{max} = 25mm $$

$f_{n}$: Frequência natural
$$ f_{n} = \frac{2}{\pi N_{a}} \frac{d}{D^2} \sqrt{ \frac{Gg}{32 \gamma} } $$

!['Densidade'](densidade.png "Densidade inox")     

In [44]:
def fn():
    g = 9.81 #m/s^2
    gamma = 0.28/27680 #convertendo para kg/m^3
    return round(( (2*(d/1000))/(pi*na()*((D(C, d)/1000)**2)) )*( (G*g/(32*gamma))**(0.5) ), 3)

fn_rpm = round(fn()*60, 3)

fn(), fn_rpm 

(4198.89, 251933.4)

$$ f_{n} = 4198,89 Hz \approx 251933,4 rpm $$

In [45]:
def r_rpm():
    return round(fn_rpm/n, 2)

r_rpm()


503.87

$$ f_{n}/n = \frac{251933,4}{500} \approx 503,87 $$

In [46]:
import pandas as pd

In [66]:
data = {
    'C': [C],
    'd [mm]': d,
    'D [mm]': D(C, d),
    '𝜏i [MPa]': ti(C),
    'Ks': ks(C),
    'Fi [N]': fi(C, d),
    'Fa [N]': fa(),
    'Fm [N]': fm(),
    'Tm [N]': tm(C, d),
    '𝜏a [MPa]': ta(C),
    'Sut [MPa]': sut(A, m),
    'Sus [MPa]': sus(A, m),
    'Sys [MPa]': sys(A, m),
    'Ses [MPa]': ses(A, m),
    '𝜏min [MPa]': tmin(fmin, C),
    'ηfs': nfs(A, m, C, d, fmin),
    'Kb': kb(),
    'σa [MPa]': sa(),
    'σm [MPa]': sm(),
    'σmin [MPa]': smin(),
    'Se [MPa]': se(A, m),
    'ηfb': nfb(),
    '𝜏ba [MPa]': tba(),
    '𝜏bm [MPa]': tbm(),
    '𝜏bmin [MPa]': tbmin(),
    'ηfsb': nfsb(),
    'k [N/mm]': k(),
    'Na [espiras]': na(),
    'Nt [espiras]': nt(),
    'Lb [mm]': lb(),
    'Lf [mm]': lf(),
    'ymax [mm]': ymax(),
    'fn [Hz]': fn(),
    'razao': r_rpm()

}

# Resultados

In [67]:
df = pd.DataFrame(data).round(2)
display(df.T)

0
C                 9.00
d [mm]            9.00
D [mm]           81.00
𝜏i [MPa]      26473.28
Ks                1.06
Fi [N]          611.05
Fa [N]          150.00
Fm [N]          850.00
Tm [N]          253.97
𝜏a [MPa]         49.32
Sut [MPa]      1354.64
Sus [MPa]       903.55
Sys [MPa]       609.59
Ses [MPa]       187.28
𝜏min [MPa]      228.07
ηfs               2.56
Kb                0.99
σa [MPa]         86.05
σm [MPa]        487.63
σmin [MPa]      401.58
Se [MPa]        279.53
ηfb               1.89
𝜏ba [MPa]        50.42
𝜏bm [MPa]       285.71
𝜏bmin [MPa]     235.30
ηfsb              2.56
k [N/mm]         12.00
Na [espiras]     10.00
Nt [espiras]     11.00
Lb [mm]          99.00
Lf [mm]         243.00
ymax [mm]        25.00
fn [Hz]        4198.89
razao           503.87

Para o problema proposto a solução encontrada foi supondo valores iniciais de indice de mola e o diâmetro da espira, assim, encontrando as dimensões faltantes dado o material escolhido, e com essas condições, chegamos a fatores de projeto para torção, flexão e fadiga de 2,56, 1,89 e 2,56 respectivamente. Na descrição do problema foi proposto um fator de projeto de 2, porém, chegamos a valores próximos utilizando as estimativas iniciais e, mais importante, valores superiores a 1, para a utilização do fator de projeto 2 seria necessario alterar as equações de coeficiente de segurança deixando explicito os valores desconhecidos, que no nosso caso foram supostos inicialmente, sendo assim, não havendo a necessidade de um chute inicial. 